test/unit/snowflake_test.js

```diff
@@ -936,6 +936,96 @@ describe('connection.execute() statement failure', function ()
   });
 });
 
+describe('connection.execute() with requestId', function () {
+  const connection = snowflake.createConnection(connectionOptions);
+  const sqlText = 'select 1;';
+  const requestId = 'SNOW-728803-requestId';
+
+  before(function (done) {
+    async.series(
+      [
+        function (callback) {
+          connection.connect(function (err, conn) {
+            assert.ok(!err, 'there should be no error');
+            assert.strictEqual(conn, connection,
+              'the connect() callback should be invoked with the statement');
+
+            callback();
+          });
+        }
+      ],
+      done
+    );
+  });
+
+  it('keep original sqlText when resubmitting requests', function (done) {
+    let statement;
+
+    async.series(
+      [
+        function (callback) {
+          // request with sqlText and requestId specified
+          statement = connection.execute(
+            {
+              sqlText: sqlText,
+              requestId: requestId,
+              complete: function (err, stmt) {
+                // if there's an error, fail the test with the error
+                if (err) {
+                  callback(err);
+                }
+                else {
+                  assert.ok(!err, 'there should be no error');
+                  assert.strictEqual(stmt, statement,
+                    'the execute() callback should be invoked with the statement');
+
+                  callback();
+                }
+              }
+            });
+
+          // the sql text and request id should be the same as what was passed
+          // in
+          assert.strictEqual(statement.getSqlText(), sqlText);
+          assert.strictEqual(statement.getRequestId(), requestId);
+        }
+      ],
+      function (err) {
+        done(err);
+      });
+  });
+
+  it('sqlText is overwritten when resubmitting requests', function (done) {
+    let statement;
+
+    async.series(
+      [
+        function (callback) {
+          // request with only requestId specified
+          statement = connection.execute(
```

Comment:
statement could be const here (but only if we do a single statement in the test as it now, look at my other comments to this test)

Reply:
Done, the test only needs a single statement